### PR TITLE
Fix: Lighthouse Issues

### PR DESCRIPTION
**Pull request by eduardoformiga**
Depends on https://github.com/vtex/faststore/pull/1712

## What's the purpose of this pull request?

- [x] Fix LH Performance Issues.
- [x] Fix Images issues for Home, PDP and PLP.
- [x] Creates custom loader with thumbor URL and VTEX CDN.
- [x] Removes custom thumbor builder and simplifies  thumbor usage in custom loader.
- [x] Adds `deviceSizes` and `imageSizes` so that Next can create `srcSets` with custom loader.
- [x] relax `resource-summary:script:count` rule count = 20. 
- [x] Fix LH SEO Issues.
- [x] Adds min tap size to 48px instead of 32px for thumb selector from ImageGallery. (double check with @renatamottam , see the comment below)

Checks are happy
Screenshot 2023-04-21 at 17 08 01

The LH issues:
(click to see details or [see here](https://github.com/vtex-sites/starter.store/runs/12865092944))

(click to see details)

(click to see details)
Screenshot 2023-04-21 at 12 14 50

## How to test it?

Compare [the previous LH logs on starter](https://github.com/vtex-sites/starter.store/runs/12865092944) with the Issues with the current on the starter PR mentioned in the section below.

### Starters Deploy Preview

https://github.com/vtex-sites/starter.store/pull/50

## References

https://nextjs.org/docs/api-reference/next/image#device-sizes
https://nextjs.org/docs/api-reference/next.config.js/custom-image-loader-config#thumbor
https://web.dev/use-thumbor/
https://github.com/GoogleChrome/lighthouse-ci/blob/main/packages/utils/src/presets/all.js#L25
https://web.dev/use-lighthouse-for-performance-budgets/
https://developer.chrome.com/docs/lighthouse/performance/resource-summary/
https://developer.chrome.com/docs/lighthouse/performance/bootup-time/

### CODE REVIEW

File: packages/ui/src/components/atoms/Button/styles.scss [hunk 1]
```diff
@@ -100,7 +100,7 @@
 
   // Small
   --fs-button-small-padding                         : var(--fs-spacing-0) var(--fs-spacing-1);
-  --fs-button-small-min-height                      : var(--fs-spacing-5);
+  --fs-button-small-min-height                      : var(--fs-spacing-7);
```
Comment (eduardoformiga) [inline packages/ui/src/components/atoms/Button/styles.scss:103]: Fixes LH SEO.
Double check this button + Icon layout with @renatamottam 
Should we increase the Icon as well?
Screenshot 2023-04-21 at 16 20 49

File: packages/lighthouse/src/index.ts [hunk 2]
```diff
@@ -41,7 +41,7 @@ const lhConfig = ({ urls, server, assertions = {} }: Params) => {
           // Number of Requests Budgets
           'resource-summary:font:count': ['error', { maxNumericValue: 3 }],
           'resource-summary:image:count': ['error', { maxNumericValue: 20 }],
-          'resource-summary:script:count': ['error', { maxNumericValue: 15 }],
+          'resource-summary:script:count': ['error', { maxNumericValue: 20 }],
```
Comment (eduardoformiga) [inline packages/lighthouse/src/index.ts:44]: relax this rule a little

File: packages/core/src/components/ui/Image/loader.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import storeConfig from 'faststore.config'
+const THUMBOR_SERVER = `https://${storeConfig.api.storeId}.vtexassets.com`
+
+export default function customImageLoader({ src, width }) {
+  const preSizeComponents = [THUMBOR_SERVER, 'unsafe']
+
+  // proportional to the width, enter a height of 0,
+  const height = 0
```
Comment (eduardoformiga) [inline packages/core/src/components/ui/Image/loader.ts:8]: Next loader don't provides height

File: packages/core/lighthouserc.js [hunk 4]
```diff
@@ -18,5 +18,6 @@ module.exports = lhConfig({
     ],
     deprecations: 'warn',
     'resource-summary:stylesheet:count': ['warn', { maxNumericValue: 400 }],
+    'resource-summary:script:count': ['warn', { maxNumericValue: 20 }],
```
Comment (eduardoformiga) [inline packages/core/lighthouserc.js:21]: relax this rule

File: packages/core/next.config.js [hunk 5]
```diff
@@ -8,6 +8,8 @@ const nextConfig = {
   /* config options here */
   images: {
     domains: [`${storeConfig.api.storeId}.vtexassets.com`],
+    deviceSizes: [320, 768, 1280, 1440],
+    imageSizes: [34, 68, 154],
```
Comment (eduardoformiga) [inline packages/core/next.config.js:12]: https://nextjs.org/docs/api-reference/next/image#device-sizes
Aims to provides `srcSet` for Next images.

File: packages/core/src/components/ui/Image/Image.tsx [hunk 6]
```diff
@@ -1,29 +1,18 @@
...
-function Image({ src, width, height, quality, ...otherProps }: ImageProps) {
-  const { src: thumborSrc, alt } = useImage({
-    src: String(src),
-    width: Number(width),
-    height: Number(height),
-    options: quality ? ({ filters: { quality } } as ThumborOptions) : undefined,
-    ...otherProps,
-  })
-
+function Image({ loading = 'lazy', ...otherProps }: ImageProps) {
```
Comment (eduardoformiga) [inline packages/core/src/components/ui/Image/Image.tsx:9]: lazy by default

File: packages/core/src/components/ui/Image/Image.tsx [hunk 7]
```diff
@@ -1,29 +1,18 @@
...
   return (
     <NextImage
       data-fs-image
-      unoptimized
-      src={thumborSrc}
-      width={width}
-      height={height}
-      alt={alt}
+      loader={loader}
```
Comment (eduardoformiga) [inline packages/core/src/components/ui/Image/Image.tsx:13]: custom loader

File: packages/core/src/components/ui/Image/Image.tsx [hunk 8]
```diff
@@ -1,29 +1,18 @@
...
+      loading={loading}
+      priority={loading === 'eager'}
```
Comment (eduardoformiga) [inline packages/core/src/components/ui/Image/Image.tsx:15]: priority hint if eager load.

File: packages/core/src/components/ui/Image/loader.ts
Comment (eduardoformiga) [inline packages/core/src/components/ui/Image/loader.ts:1]: Merges previous Thumbor builder into Next Image custom loader .

**Review by ArthurTriis1 — APPROVED** [review]

**Comment by eduardoformiga** [comment]
Close due to problems after changing base branch.
moved to https://github.com/vtex/faststore/pull/1737

**Pull request closed without merging**